File: chess-game/src/misc/Pieces.js
export default class Pieces{
    static None = 0;
    static King = 1;
    static Pawn = 2;
    static Bishop = 3;
    static Knight = 4;
    static Rook = 5;
    static Queen = 6;

    static White = 8;
    static Black = 16;

    static getColor(piece){
        return (piece & 24);
    }

    static isWhite(piece){
        return (piece & 24) === this.White
    }

    static isBlack(piece){
        return (piece & 24) === this.Black
    }

    static isKing(piece){
        return (piece & 7) === this.King
    }

    static isPawn(piece){
        return (piece & 7) === this.Pawn
    }

    static isQueen(piece){
        return (piece & 7) === this.Queen
    }

    static isRook(piece){
        return (piece & 7) === this.Rook
    }

    static isBishop(piece){
        return (piece & 7) === this.Bishop
    }

    static isKnight(piece){
        return (piece & 7) === this.Knight
    }

    static isNone(piece){
        return piece === this.None;
    }

    static charToImage = new Map([
        ['n', "BlackKnight"],
        ['q', "BlackQueen"],
        ['b', "BlackBishop"],
        ['r', "BlackRook"],
        ['k', "BlackKing"],
        ['p', "BlackPawn"],
        ['Q', "WhiteQueen"],
        ['N', "WhiteKnight"],
        ['B', "WhiteBishop"],
        ['R', "WhiteRook"],
        ['K', "WhiteKing"],
        ['P', "WhitePawn"]
    ]);

    static charToNumber = new Map([
        ['n', this.Black | this.Knight],
        ['q', this.Black | this.Queen],
        ['b', this.Black | this.Bishop],
        ['r', this.Black | this.Rook],
        ['k', this.Black | this.King],
        ['p', this.Black | this.Pawn],
        ['Q', this.White | this.Queen],
        ['N', this.White | this.Knight],
        ['B', this.White | this.Bishop],
        ['R', this.White | this.Rook],
        ['K', this.White | this.King],
        ['P', this.White | this.Pawn]
    ]);

    static isSameColor(piece, currPlayer) {
        piece &= 24;

        return (currPlayer & piece) !== 0;
    }
}

File: chess-engine/src/main/java/com/example/chessengine/MyEngine/Pieces.java
package com.example.chessengine.MyEngine;

public class Pieces {
    public static final int None = 0;
    public static final int King = 1;
    public static final int Pawn = 2;
    public static final int Bishop = 3;
    public static final int Knight = 4;
    public static final int Rook = 5;
    public static final int Queen = 6;
    public static final int White = 8;
    public static final int Black = 16;

    public static boolean isSameColor(int color, int piece){
        color &= 24;
        piece &= 24;

        return color == piece;
    }


    public static boolean isWhite(int piece){
        return (piece & 24) == Pieces.White;
    }

    public static boolean isNone(int piece){
        return (piece & 7) == Pieces.None;
    }

    public static boolean isKing(int piece){
        return (piece & 7) == Pieces.King;
    }

    public static boolean isQueen(int piece){
        return (piece & 7) == Pieces.Queen;
    }

    public static boolean isPawn(int piece){
        return (piece & 7) == Pieces.Pawn;
    }

    public static boolean isRook(int piece){
        return (piece & 7) == Pieces.Rook;
    }

    public static boolean isBishop(int piece){
        return (piece & 7) == Pieces.Bishop;
    }

    public static boolean isKnight(int piece){
        return (piece & 7) == Pieces.Knight;
    }

    public static boolean isBishopOrQueen(int piece){
        return ((piece & 7) == Pieces.Bishop) || ((piece & 7) == Pieces.Queen);
    }

    public static boolean isRookOrQueen(int piece){
        return ((piece & 7) == Pieces.Rook) || ((piece & 7) == Pieces.Queen);
    }

    public static int getPiece(Move.Flag flag){
        return switch (flag){
            case PROMOTE_TO_QUEEN -> Queen;
            case PROMOTE_TO_ROOK -> Rook;
            case PROMOTE_TO_KNIGHT -> Knight;
            case PROMOTE_TO_BISHOP -> Bishop;
            default -> None;
        };
    }

    public static int PieceType(int piece){
        return piece & 7;
    }
}
